### PR TITLE
Second part to fully resolve #1229: convert grpc module too

### LINKED ISSUE #1229: Replace direct Guava usage by Stargate (use shaded everywhere applicable) to reduce Guava conflict w/ Cassandra backend
There was a recent issue with an old Guava version (19.0) getting included due to exclusions, transitive dependencies, causing some test setup failures. Let's add a global managed version to try to remove/reduce likelihood of such problem resurfacing.

Although the first idea was same as #814 it turned out more difficult to try to align Guava version across C* 3.11, 4.0 and DSE (impossible, basically) and it appears much better to use Datastax OSS driver shaded Guava uniformly.
This lets underlying persistence/storage to use Guava version they do without affecting Stargate.
This may mean we require newer shaded version (current one is 25), but for now the version we have works fine.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #1229

**Checklist**
- [-] Changes manually tested -- automated test suite should cover since issues would be mostly for startup (osgi)
- [-] Automated Tests added/updated -- no functional changes, existing IT/CI should be sufficient
- [-] Documentation added/updated -- no functional changes

### CODE REVIEW

File: restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java
Comment (ivansenic) [inline restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java:5]: Was Eric faster in merging the shredding than you? Can we ensure once we merge this that it's top of the master and there are no direct usages of guava?

**Review by ivansenic — COMMENTED** [review]
Looks good so far..

File: restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java
Comment (tatu-at-datastax) [reply restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java:5]: I am not 100% sure how this came about. I will make sure to merge from master and check that the only Guava usage is from `persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/StargateSystemKeyspace.java` (which is due to Cassandra leaking `ListenableFuture`)

**Comment by tatu-at-datastax** [comment]
Should be good, now with DSE persistence too. Waiting for CI to complete; will keep in-sync with master as well.

**Review by dougwettlaufer — APPROVED** [review]

**Review by tomekl007 — APPROVED** [review]

**Review by ivansenic — APPROVED** [review]
Perfect..

**Pull request merged**
